### PR TITLE
Topic recognition - MRI Super Resolution Network (Elijah Nelson 46991553)

### PR DESCRIPTION
**Pull request by elijah-nelson**
# MRI Super Resolution Network (Elijah Nelson 46991553)
### Abstract
This project implements a brain MRI super-resolution network by training on the ADNI brain dataset. The network is trained to up-scale from 4x down-sampled input images and produce reasonably clear output images.

As proposed by [Shi, 2016](https://arxiv.org/abs/1609.05158), the model utilises efficient sub-pixel convolution by extracting feature maps in the low-resolution space. This reduces computational complexity by waiting until the end of the model to reconstruct to higher dimensions.

An example of the model upsampling a low-resolution image to a higher resolution image:

### Added files
    doc/        # Folder for images used in the README
    imgs/       # Folder that generated images are saved to
    config.py   # Hyperparameters and other constants
    dataset.py  # Data transformation, preprocessing and loading
    generate.py # Model output generation
    modules.py  # Model architecture definition
    predict.py  # Example model usage
    train.py    # Model training and validation

### Dependencies

This project requires the following packages:

 - pytorch: v2.0.1
 - matplotlib: v3.7.1

### CODE REVIEW

**Comment by uqmarlonbran** [comment]
**This is an initial inspection - no action is required yet**

**Difficulty:** Normal

**General Comments:** Solves the problem overall. Improves image quality compared to low-rez images. However, no validation set was used during training. Simply stores the last version of the network at the end of training. 

**[Recognition Problem]**
"modules.py":

- Correct implementation of the network, though seems they're using 3 channels rather than 1 grayscale one.
- Model architecture mostly the same as PixelCNN, just a few more convolutional layers.
- Allows for variable upscaling factor.

"dataset.py":

- Uses ImageFolder to load images, though, it requires the user to insert another folder into the file structure (at least this is documented in the README)
- Otherwise seems ok, nice check for dimensionality.

"train.py":

- Correctly tracks losses during training.
- No validation set (though it is claimed in the readme).
- Final network is saved.

"predict.py":

- Generates some images from the test dataset, though no metrics are performed.

"README.MD":

- Provides a brief description of the network architecture.
- Demonstrates some input and output images.
- Provides training loss but no validation loss or evaluation of metrics on the test set.
- Reconstructions seem of higher quality than the low rez versions.
- Details expected directory structure.
- Has dependencies list.

**[Commit Log]**
Commit log looks ok.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no validation set -1
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable but no usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
